### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by che30**
In this project I carried out the following: 
- A class was created for a project 
- A class for a task to be added
- A class for the user interface
- Methods for functionality to add a project, add a task  delete a project and task were added
- Linters was set up
Any feedback and proposition is highly welcomed

### CODE REVIEW

**Review by teekaytech — CHANGES_REQUESTED** [review]
### Status: Invalid Code Review Request :x: 

__Hi @che30, I commend you for the great effort put into this project. However, I have to mark the project as invalid due to the following reason(s):__

- [ ] Your GitHub Action checks are not running. For your project to be considered for review, you must ensure that all GitHub Action checks are running successfully with no errors.

_**This submission will be marked as invalid. Kindly ensure you have running and passing GitHub Action checks before you submit another review request.**_



* * *

**_You can reach out to me on Twitter (@ola_lere), Github(@teekaytech), or slack (@taofeekolalere )_**

Happy Coding :man_technologist:

File: README.md [hunk 1]
```diff
@@ -1 +1,48 @@
-# TodoList
\ No newline at end of file
+# TodoList
+This is a project about to-do's . Projects can be added and tasks(to-do's) added under each project.  The priorities chosen will display a different background color for any todo(task)  The to-do's and projects will be stored on local storage.
+ 
+![screenshot](./dist/assets/images/todoshotOne.png)
```
Comment (christianotieno) [inline README.md:4]: ## Improve the screenshot.
- As part of having and maintaining a professional repo, the README, being the gateway to the project ought to stand out at all time as it is what sells the project the most. 
- [x] Please ensure that the screenshot is showing here as expected.

File: README.md [hunk 2]
```diff
@@ -1 +1,48 @@
...
+- BOOTSTRAP FRAMEWORK
+- Javasript
+
+## Getting started
+   To get a local copy up and running follow these simple example steps.
+# Setup
+- Clone the project locally.
+- Type the command `cd TodoList`
+- run the command   `yarn install`
+- Use any browser of your choice, copy the absolute path of the  `index.html` and type in the search bar
```
Comment (christianotieno) [inline README.md]: ## Improve the instructions of cloning and running the app locally.
- As part of having and maintaining a professional repo, the README, being the gateway to the project ought to stand out at all time as it is what sells the project the most.
- [x] Do you mind adding more instructions for running the code locally using webpack and the set scripts, in addition to the cloning instructions?

File: README.md [hunk 3]
```diff
@@ -1 +1,48 @@
...
+
+- [Live Demo Link](https://che30.github.io/TodoList/)
+ 
+
+## Author
+**Che Blanchard**
+
+- GitHub: [@che30](https://github.com/che30)
+- LinkedIn: [Che Blanchard](https://www.linkedin.com/in/che-nsoh-9455271b0/)
+- Twitter:[che55085128](https://twitter.com/che55085128)
```
Comment (christianotieno) [inline README.md]:
```suggestion
- GitHub: [@che30](https://github.com/che30)
- LinkedIn: [Che Blanchard](https://www.linkedin.com/in/che-nsoh-9455271b0/)
- Twitter: [@che55085128](https://twitter.com/che55085128)
```

File: README.md [hunk 4]
```diff
@@ -1 +1,48 @@
...
+
+Contributions, issues, and feature requests are welcome!
+
+## Show your support
+
+Give a ⭐️ if you like this project!
+
+## License
+
+This project is [MIT](./LICENSE.txt) licensed.
```
Comment (christianotieno) [inline README.md]: ## Fix or remove dead links.
- [x] the license link is currently a dead link, please fix this by either adding a working link to the project's licence or remove the link for a much cleaner README, thank you.

File: dist/assets/stylesheets/main.css [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+  position: fixed; /* Stay in place */
+  z-index: 1; /* Sit on top */
+  padding-top: 50px; /* Location of the box */
+  left: 0;
+  top: 0;
+  width: 100%; /* Full width */
+  height: 100%; /* Full height */
+  overflow: auto; /* Enable scroll if needed */
+  background-color: rgb(0,0,0); /* Fallback color */
+  background-color: rgba(0,0,0,0.4); /* Black w/ opacity */
```
Comment (christianotieno) [inline dist/assets/stylesheets/main.css]: ## Aim to remain `DRY`
- Though having comments on what the code is doing is highly encouraged. it is also encouraged to aim at `self-documenting` code. Thereby it relieves you the burden of getting to write unnecessary comments.   
- [x] In this instance, having these comments here feel like you are repeating yourself as the comments are doing exactly what the code demonstrates. I urge we remove most of them to have a cleaner codebase. Like so 👇🏾 
```suggestion
  display: none; 
  position: fixed; 
  z-index: 1; 
  padding-top: 50px; 
  left: 0;
  top: 0;
  width: 100%; 
  height: 100%; 
  overflow: auto;
  background-color: rgb(0,0,0); 
  background-color: rgba(0,0,0,0.4); 
```

File: dist/assets/stylesheets/main.css [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+.add-task {
+  cursor: pointer;
+  margin-left: 20%;
+  margin-top: 15px;
+}
+/* .todo-form .input-group {
+  display: flex;
+  flex-direction: column;
+  margin-bottom: 18px;
+} */
```
Comment (christianotieno) [inline dist/assets/stylesheets/main.css]: ## Avoid committing and pushing commented out code.
- [ ] It’s better to delete **commented piece of codes**. If you need to review a previous version of your code, use `git history` instead. More on this on this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/).

File: src/Task.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    return tasks;
+  }
+
+  static storeTask(task) {
+    const tasks = Task.getTask();
+    tasks.push(task);
+    localStorage.setItem('tasks', JSON.stringify(tasks));
+  }
+}
+//  localStorage.clear()
```
Comment (christianotieno) [inline src/Task.js]: ## Avoid committing and pushing commented out code.
- [x] It’s better to delete **commented piece of codes**. If you need to review a previous version of your code, use `git history` instead. More on this on this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/).

File: src/index.js [hunk 8]
```diff
@@ -0,0 +1,121 @@
+
```
Comment (christianotieno) [inline src/index.js:1]: ## Optional change
- [ ] Can we have a scenario where after creating a project or a todo, the forms get removed on save without the user needing to reload the page? The problem appears to be there when moving from project to project as well; the user gets forced to reload the page to see the previously created todos. Do you mind fixing this?
- [ ] Also can we have the error message actually showing on the form fields when the user attempts to submit a half-filled/empty form? Currently, the rules have been enforced but the user cant sees them on the page.

File: src/index.js [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+        priority,
+        Date.now().toString(),
+        giveStoredProjects[Number(giveMeActiveProject[0])].name);
+      Task.storeTask(task);
+    }
+    UI.showAlert('Task added successfuly', 'success');
+    UI.delay();
+  }
+});
+//  UI.renderDefault();
```
Comment (christianotieno) [inline src/index.js]: ## Avoid committing and pushing commented out code.
- [ ] It’s better to delete **commented piece of codes**. If you need to review a previous version of your code, use `git history` instead. More on this on this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/).

**Review by christianotieno — CHANGES_REQUESTED** [review]
Carried out mandatory changes as requested by TSE

**Review by voscarmv — CHANGES_REQUESTED** [review]
# Impressive work. 💻 👀 ✨ 
- Good work applying the changes requested by the previous code reviewer.
## However, there are some changes required 👀 🧰 🔨 🔧 
- When I add new ToDos to the default project, and then I click on the `default` project link, some ToDos are unintenionally deleted. Please fix this bug.
- When I add new Projects after and add some ToDos to the `default` as well as to the new project, my ToDos get deleted for no reason when I switch between different projects by clicking on their links. Please fix this as well.
- The ToDos are not displaying their priority level. Please show the priority level and also display it on the Edit form.
### Submit these changes and request a new code review! 😄 👍 

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️

File: dist/assets/stylesheets/main.css [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+.dropbtn {
+  background-color: #131413;
+  color: white;
+  padding: 16px;
+  font-size: 16px;
+  border: none;
+  width: 210%;
+}
+.display-3{
+  /* margin-left: 30%; */
```
Comment (Forison) [inline dist/assets/stylesheets/main.css]: * [x] Remove commented code

File: src/UI.js [hunk 11]
```diff
@@ -0,0 +1,215 @@
+import Project from './Project';
+
+export default class UI {
+  static delay() {
+    setTimeout(() => {
+      window.location.reload();
+    }, 3000);
+  }
+
+  //  static setSelected(){
```
Comment (Forison) [inline src/UI.js]: * [x]  Remove commented code

File: src/UI.js [hunk 12]
```diff
@@ -0,0 +1,215 @@
...
+  static getProjectName() {
+    let projects;
+    if (localStorage.getItem('projects') === null) {
+      projects = [];
+    } else {
+      projects = JSON.parse(localStorage.getItem('projects'));
+    }
+    return projects;
+  }
+  //  static getTargettedElement(){
```
Comment (Forison) [inline src/UI.js]: * [x]  Remove commented code

File: src/index.js [hunk 13]
```diff
@@ -0,0 +1,153 @@
...
+  currentNumber = Number(lastindex);
+  localStorage.setItem('lastProjectId', JSON.stringify(currentNumber));
+  localStorage.setItem('projects', JSON.stringify(projects));
+  UI.addProjectName();
+  UI.showAlert('Project Removed', 'success');
+  UI.delay();
+};
+window.removeTask = (taskNumber) => {
+  const tasks = Task.getTask();
+  // const selectedId = JSON.parse(localStorage.getItem('selectedId'));
```
Comment (Forison) [inline src/index.js]: * [x]  Remove commented code

File: src/index.js [hunk 14]
```diff
@@ -0,0 +1,153 @@
...
+  });
+
+
+  tasks.forEach((task, index) => {
+    if (Number(task.now) === taskNumber) {
+      tasks.splice(index, 1);
+      document.getElementById(taskNumber).parentNode.parentNode.parentNode.remove();
+      localStorage.setItem('tasks', JSON.stringify(tasks));
+      UI.showAlert('Task Removed', 'success');
+      // const selectedId = JSON.parse(localStorage.getItem('selectedId'));
```
Comment (Forison) [inline src/index.js]: * [x]  Remove commented code

File: src/index.js [hunk 15]
```diff
@@ -0,0 +1,153 @@
...
+  let toBeEdited;
+  const tasks = Task.getTask();
+  tasks.forEach((task, index) => {
+    if (Number(task.now) === taskNumber) {
+      toBeEdited = task;
+      tasks.splice(index, 1);
+      localStorage.setItem('tasks', JSON.stringify(tasks));
+    }
+  });
+  // const selectedId = JSON.parse(localStorage.getItem('selectedId'));
```
Comment (Forison) [inline src/index.js]: * [x]  Remove commented code

File: src/index.js [hunk 16]
```diff
@@ -0,0 +1,153 @@
...
+      Task.storeTask(task);
+      projects[Number(giveMeActiveProject[0])].tasks.push(task);
+      localStorage.setItem('projects', JSON.stringify(projects));
+    }
+    UI.showAlert('Task added successfuly', 'success');
+    UI.delay();
+  }
+});
+
+//  const selectedId = UI.getSelected()
```
Comment (Forison) [inline src/index.js]: * [x]  Remove commented code

**Review by Forison — CHANGES_REQUESTED** [review]
# Status::Approved✔️

#### You have done a good job on this one so far, all the issues raised in the last review have been addressed😶.

* [x]  Currently, todo gets deleted after I update, let address this issue.
* [x] Check other changes to be effected below👇

#### That is it, kindly make the changes and submit a re-review💪

#### Happy coding👨‍💻

Addo Forison

**Comment by che30** [comment]
Can't seem to find any commented code again sir

**Review by christianotieno — APPROVED** [review]
## Project Approved ♻️ 
Hello @che30 👋🏾 
Good job working on the changes. This project is now approved. 🥇 
### Happy coding 💻 
#### Christian Otieno (CR)

**Pull request merged**